Source/WebCore/accessibility/AccessibilityScrollView.cpp

```diff
@@ -338,15 +338,20 @@ AccessibilityObject* AccessibilityScrollView::parentObject() const
     if (!cache)
         return nullptr;
 
-    WeakPtr owner = m_frameOwnerElement.get();
+    Element* parentElement = m_frameOwnerElement.get();
     if (RefPtr localFrameView = dynamicDowncast<LocalFrameView>(m_scrollView.get()))
-        owner = localFrameView->frame().ownerElement();
+        parentElement = localFrameView->frame().ownerElement();
     else if (RefPtr remoteFrameView = dynamicDowncast<RemoteFrameView>(m_scrollView.get()))
-        owner = remoteFrameView->frame().ownerElement();
-
-    if (owner && owner->renderer())
-        return cache->getOrCreate(*owner);
-    return nullptr;
+        parentElement = remoteFrameView->frame().ownerElement();
+
+    AccessibilityObject* parentAccessibilityObject = nullptr;
+    while (parentElement && !parentAccessibilityObject) {
+        parentAccessibilityObject = cache->getOrCreate(*parentElement);
+        if (parentAccessibilityObject)
+            break;
```

Comment:
Feel free to not do with this if you think the way you've written is cleaner. But a way to do this in one line would be:

```cpp
if ((ancestorAccessibilityObject = cache->getOrCreate(*ancestorElement))
    break;
```